pkgs/servers/openvscode-server/default.nix

```diff
@@ -133,113 +123,86 @@ stdenv.mkDerivation (finalAttrs: {
+    '';
 
-      # put ripgrep binary into bin so postinstall does not try to download it
+  configurePhase =
+    ''
+      export HOME=$TMPDIR/home
+      mkdir -p $HOME
+    ''
+    ## unpack all of the prefetched node_modules folders
+    + ''
+      for p in $(find -name package.json -exec dirname {} \;)
+      do (
+        cd $p
+        if [ -e node_modules ]
+        then
+          echo >&2 "File exists $p/node_modules"
+          exit 0
+        fi
+        cp -r $nodeModules/$p/node_modules .
+        chmod -R 700 node_modules
+      )
+      patchShebangs $p/node_modules
+      done
+    ''
+    ## put ripgrep binary into bin so postinstall does not try to download it
+    + ''
       find -path "*@vscode/ripgrep" -type d \
         -execdir mkdir -p {}/bin \; \
         -execdir ln -s ${ripgrep}/bin/rg {}/bin/rg \;
     ''
-    + lib.optionalString stdenv.hostPlatform.isDarwin ''
-      # use prebuilt binary for @parcel/watcher, which requires macOS SDK 10.13+
-      # (see issue #101229)
-      pushd ./remote/node_modules/@parcel/watcher
-      mkdir -p ./build/Release
-      mv ./prebuilds/darwin-x64/node.napi.glibc.node ./build/Release/watcher.node
-      jq "del(.scripts) | .gypfile = false" ./package.json | sponge ./package.json
-      popd
+    ## pre-seed node-gyp
+    + ''
+      mkdir -p $HOME/.node-gyp/${nodejs.version}
+      echo 11 > $HOME/.node-gyp/${nodejs.version}/installVersion
+      ln -sfv ${nodejs}/include $HOME/.node-gyp/${nodejs.version}
     ''
+    ## rebuild native binaries
     + ''
-      export NODE_OPTIONS=--openssl-legacy-provider
-
-      # rebuild binaries, we use npm here, as yarn does not provide an alternative
-      # that would not attempt to try to reinstall everything and break our
-      # patching attempts
+      echo >&2 "Rebuilding from source in ./remote"
       npm --prefix ./remote rebuild --build-from-source
-
-      # run postinstall scripts after patching
-      find . -path "*node_modules" -prune -o \
-        -path "./*/*" -name "yarn.lock" -printf "%h\n" | \
-          xargs -I {} sh -c 'jq -e ".scripts.postinstall" {}/package.json >/dev/null && yarn --cwd {} postinstall --frozen-lockfile --offline || true'
-
-      # build and minify
-      yarn --offline gulp vscode-reh-web-${vsBuildTarget}-min
-
-      runHook postBuild
+    ''
+    ## run postinstall scripts
+    + ''
+      find -name package.json -type f -exec sh -c '
+        if jq -e ".scripts.postinstall" {} >-
+        then
+          echo >&2 "Running postinstall script in $(dirname {})"
+          npm --offline --prefix=$(dirname {}) run postinstall
+        fi
+        exit 0
+      ' \;
     '';
 
-  installPhase = ''
-    runHook preInstall
+  buildPhase = ''
+    npm run gulp vscode-reh-web-${vsBuildTarget}-min
+  '';
 
+  installPhase = ''
     mkdir -p $out
     cp -R -T ../vscode-reh-web-${vsBuildTarget} $out
-    ln -s ${nodejs}/bin/node $out
-
-    runHook postInstall
+    ln -sf ${nodejs}/bin/node $out
```

Comment:
Why delete `runHook` lines?

Reply:
same reason I deleted the esbuild patching: I don't think they are necessary. nothing sets a hook, the build seemed fine without them, so I applied occam's razor.

are you saying pre- and post-hooks should always be run in custom phases?

Reply:
Yes! See:

https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/missing-phase-hooks.md

Reply:
I'm not convinced that making contributors add boilerplate is a good way forward here. If necessary, I'd much rather have something like [this comment](https://github.com/NixOS/nixpkgs/issues/175605#issuecomment-1381007025) from the (as yet unresolved) underlying issue suggests: Make `runHook` idempotent and always run hooks in the default builder.

Reply:
@bendlas good point. but for now, I hope you can add them so we can maybe get this backported to 25.05?